### PR TITLE
Backport gradle changes made in #22371

### PR DESCRIPTION
**Pull request by javanna**
This is the backport of the gradle changes made as part of #22371.

Backporting these gradle changes will make backporting the high level rest client submodule easier in the future when ready (it is currently under development in master), by simply copying its directory from master to 5.x ideally.

### CODE REVIEW

File: distribution/build.gradle [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Licensed to Elasticsearch under one or more contributor
+ // * Licensed to Elasticsearch under one or more contributor
```
Comment (nik9000) [inline distribution/build.gradle]: Leftover?

File: qa/smoke-test-http/src/test/java/org/elasticsearch/http/ContextAndHeaderTransportIT.java [hunk 2]
```diff
@@ -322,7 +321,7 @@ public int order() {
         }
 
         @Override
-        protected boolean apply(String action, ActionRequest request, ActionListener listener) {
+        protected boolean apply(String action, ActionRequest request, ActionListener<?> listener) {
```
Comment (nik9000) [inline qa/smoke-test-http/src/test/java/org/elasticsearch/http/ContextAndHeaderTransportIT.java]: Leftover? If it compiles I'm happy to include it.

**Review by nik9000 — APPROVED** [review]
Feels weird to be reviewing my own code, but it is your backport. Looks good to me.

File: qa/smoke-test-http/src/test/java/org/elasticsearch/http/ContextAndHeaderTransportIT.java (hunk 2 above)
Comment (javanna) [inline qa/smoke-test-http/src/test/java/org/elasticsearch/http/ContextAndHeaderTransportIT.java]: I think it is not required yet but better anyways.

File: distribution/build.gradle (hunk 1 above)
Comment (javanna) [inline distribution/build.gradle]: ops. I pushed this to master too :( will fix

**Comment by javanna** [comment]
> Feels weird to be reviewing my own code, but it is your backport. Looks good to me.

Right, I made a PR mainly for proper labelling and clarity. And to make sure I don't make mistakes :) Thanks for looking at your own code once again.

**Pull request merged**